core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -581,6 +640,18 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    */
   final def interruptible: ZIO[R, E, A] = interruptStatus(InterruptStatus.Interruptible)
 
+  /**
+   * Returns an effect that when interrupted returns immediately.
+   * However, the effect and its interruption occurs in a forked fiber
+   */
+  final def interruptibleFork: ZIO[R, E, A] =
+    ZIO.uninterruptible {
+      for {
+        fiber <- self.interruptible.fork.daemon
+        res   <- fiber.join.interruptible.onInterrupt(fiber.interrupt.fork.daemon)
```

Comment:
@dariusrobson 
Using `.interruptible` precludes code from being usable inside finalizers, see: https://github.com/zio/zio/issues/2005
Usage here can be replaced by `uninterruptibleMask` (`.interruptible.fork` should remain as is though, because that `.interruptible` is in a different fiber.)

Reply:
This is a `fork` version of `interruptible`, so it should work exactly the same as `interruptible`. If it were implemented using `uninterruptibleMask`, then it would not yield expected results. If we end up deleting `interruptible`, we would delete `interruptibleFork` along with it.

Reply:
@jdegoes 
Sorry, I was confused about the intended semantics here.

>  If we end up deleting `interruptible`, we would delete `interruptibleFork` along with it.

tbh, I'd think that a function to fork off/skip waiting for an inner action's interruption is useful independently from `interruptible` and can be orthogonal to setting a fiber's interruptibility status. I think it may be useful in general to be able to specify *how* interruption finalization will execute, orthogonally from specifying whether it *can* happen, e.g. `.onInterrupt` does not force a fiber to be interruptible, even though it's a no-op in an uninterruptible section.

Reply:
I agree with you that such an independent action would be quite nice. 

I think interruption finalizers have to operate the same as others, since internally they are erased (there are no finalizers in ZIO core); also violating linearity of "finalizers" would have grave consequences for reasoning about code. I'm not sure we have much choice there.

But basically in any case where interruption is happening, we need to be able to "choose" whether to wait for that or to fork it. I think there's probably a better way to do that than myriad `fork` variants, but we probably (and unfortunately) aren't going to figure that out pre 1.0...

Reply:
@jdegoes 
By interruption finalizaters I mean just the usual finalization - when it's triggered by external interruption, i.e. just having a variant of `interruptibleFork` without `interruptible` would allow choosing whether to fork out this finalization or not without affecting the interruptible flag.

Reply:
Yes, I like that.